vm/src/builtins/dict.rs

```diff
@@ -998,23 +999,33 @@ trait ViewSetOps: DictView {
     ) -> PyResult<PyComparisonValue> {
         match_class!(match other {
             ref dictview @ Self => {
-                PyDict::inner_cmp(
+                return PyDict::inner_cmp(
                     zelf.dict(),
                     dictview.dict(),
                     op,
                     !zelf.class().is(vm.ctx.types.dict_keys_type),
                     vm,
-                )
+                );
             }
             ref _set @ PySet => {
                 let inner = Self::to_set(zelf.to_owned(), vm)?;
                 let zelf_set = PySet { inner }.into_pyobject(vm);
-                PySet::cmp(zelf_set.downcast_ref().unwrap(), other, op, vm)
+                return PySet::cmp(zelf_set.downcast_ref().unwrap(), other, op, vm);
             }
+            ref _dictitems @ PyDictItems => {}
+            ref _dictkeys @ PyDictKeys => {}
             _ => {
-                Ok(NotImplemented)
+                return Ok(NotImplemented);
             }
-        })
+        });
+        let self_vec: Vec<PyObjectRef> = zelf.as_object().to_owned().try_into_value(vm)?;
+        let other_vec: Vec<PyObjectRef> = other.to_owned().try_into_value(vm)?;
+        let res = self_vec.eq(vm, &other_vec)?;
+        match op {
+            PyComparisonOp::Eq => Ok(Implemented(res)),
+            PyComparisonOp::Ne => Ok(Implemented(!res)),
+            _ => Ok(NotImplemented),
+        }
```

Comment:
I revised `ObjectSequenceOp` to `PyExactSizeIterator` in  #3785

For simple fix, this version will work:
```suggestion
        # TODO: Check len() before creating Vec
        let self_vec: Vec<PyObjectRef> = zelf.as_object().to_owned().try_into_value(vm)?;
        let other_vec: Vec<PyObjectRef> = other.to_owned().try_into_value(vm)?;
        self_vec.iter().cmp(other_vec.iter(), op, vm)
```

Could you rebase it on current main?

If you are interested in removing eager `Vec` collecting, you can try to implement `PyExactSizeIterator` trait on `PyDictItems` and  `PyDictKeys`.

Reply:
I reflected it as you said. Thanks! :D